README.md

```diff
@@ -142,6 +142,32 @@ smb2Client.rename('path\\to\\my\\file.txt', 'new\\path\\to\\my\\new-file-name.tx
 ### smb2Client.close ( )
 This function will close the open connection if opened, it will be called automatically after ```autoCloseTimeout``` ms of no SMB2 call on the server.
 
+### smb2Client.createReadStream ( fileName, [options], callback )
+Returns a read stream on the file. Unlike fs.createReadStream, this function is asynchronous, as we need use asynchronous smb requests to get the stream.
+
```

Comment:
fs.createReadStream manages to be 'synchronous' not because the files are local, but because the stream interface takes care of all the asynchronicity that's needed. A ReadableStream is a perfectly appropriate place to emit an error, even an error from the smb request used to setup the stream. This would be comparable to getting a permission error emitted from an fs.createReadStream().

The asynchronous nature of the setup does not necessitate a callback. I recommend matching the interface of fs.createReadStream. If I understand es7 correctly, you should be able to 'await request("open"…)' in SmbReadableStream.prototype._read().

createWriteStream should be handled similarly.